nixos/modules/virtualisation/qemu-vm.nix

```diff
@@ -111,14 +111,12 @@ let
           ${if (cfg.useBootLoader && cfg.useDefaultFilesystems) then ''
             # Create a writable qcow2 image using the systemImage as a backing
             # image.
-
-            # CoW prevent size to be attributed to an image.
-            # FIXME: raise this issue to upstream.
             ${qemu}/bin/qemu-img create \
               -f qcow2 \
               -b ${systemImage}/nixos.qcow2 \
               -F qcow2 \
-              "$NIX_DISK_IMAGE"
+              "$NIX_DISK_IMAGE" \
+              "${toString cfg.diskSize}M"
```

Comment:
default diskSize is 1024, which can cause to small images.
I would keep the old behaviour if diskSize is auto

Reply:
qemu images in nixpkgs do not support an "auto" value, they do need a a static size.